state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -773,13 +773,18 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		if let Some(broadcast_data) = AwaitingBroadcast::<T, I>::get(broadcast_id) {
 			// If the broadcast is not pending, we should not retry.
-			if let Some(api_call) = PendingApiCalls::<T, I>::get(broadcast_id) {
-				if T::TransactionBuilder::requires_signature_refresh(
-					&api_call,
-					&broadcast_data.threshold_signature_payload,
-					CurrentOnChainKey::<T, I>::get(),
-				) {
+			if let Some(mut api_call) = PendingApiCalls::<T, I>::get(broadcast_id) {
+				if let RequiresSignatureRefresh::True(maybe_modified_apicall) =
+					T::TransactionBuilder::requires_signature_refresh(
+						&api_call,
+						&broadcast_data.threshold_signature_payload,
+						CurrentOnChainKey::<T, I>::get(),
+					) {
 					Self::deposit_event(Event::<T, I>::ThresholdSignatureInvalid { broadcast_id });
+					if let Some(modified_apicall) = maybe_modified_apicall {
+						PendingApiCalls::<T, I>::insert(broadcast_id, modified_apicall.clone());
+						api_call = modified_apicall;
```

Comment:
Do we not need to replace the PendingApiCall?

Reply:
We are doing that in the line above. I dont understand?

Reply:
I missed that you're right 😅 I was looking for it further down.